src/base/bittorrent/session.cpp

```diff
@@ -3771,6 +3795,9 @@ void Session::refresh()
 {
     m_nativeSession->post_torrent_updates();
     m_nativeSession->post_session_stats();
+    updateTorrentStatusReport();
```

Comment:
Why do you want to update it here? Isn't it more logical to do this in an existing place once torrents are updated? Besides, isn't it obvious that no statistics (I mean torrent statuses that recalculated in `updateTorrentStatusReport()`) have changed if no torrent has changed? Why then send a signal about a meaningless event?

Reply:
>Besides, isn't it obvious that no statistics (I mean torrent statuses that recalculated in updateTorrentStatusReport()) have changed if no torrent has changed? 

You mean in `handleStateUpdateAlert()` right? Then no, it is not fully correct to place it there, the fact that the calculation has worked is because the regular timeout, not because of torrent state updated. For example when you delete a torrent, `lt::state_update_alert *p` won't contain about any torrent state has updated, i.e. you need more/other alerts to make sure the calculation will be right. So currently `updateTorrentStatusReport()` is operating in a polling fashion and hence I moved it here.

>Why then send a signal about a meaningless event?

Until every thing is event based, you still need a regular interval heartbeat. And so far `torrentsUpdated()` signal isn't equal to `refreshed()` signal IMO.